include/jemalloc/internal/rtree_types.h

```diff
@@ -25,11 +26,17 @@ typedef struct rtree_s rtree_t;
 #define RTREE_HEIGHT_MAX						\
     ((1U << (LG_SIZEOF_PTR+3)) / RTREE_BITS_PER_LEVEL)
 
+/*
+ * Number of leafkey/leaf pairs to cache.  Each entry supports an entire leaf,
+ * so the cache hit rate is typically high even with a small number of entries.
+ * Two entries is enough to keep the hit rate high even when an arena uses a
+ * combination of dss and mmap.
+ */
+#define RTREE_CTX_NCACHE 2
```

Comment:
I did a bunch of benchmarks prior to converting the LRU cache from element to leaf caching, and 2 was the largest number that didn't show a substantial negative impact when suffering a high miss rate.  After converting to a leaf cache, I measured very little sensitivity to this setting, because the hit rate is very high in practice.  We could perhaps bump this up as high as ~8, but beyond that we might suffer tsd cacheline misses instead of rtree cacheline misses, negating any benefit.